### DISCUSSION TITLE
Message ordering

### DISCUSSION THREAD

**Discussion by NikitaMishin** [uNetworking/uWebSockets#1315] [Q&A]
Is uWebSockets guarantee the preserving ordering of sending and receiving messages?
e.g if some client send to us packet A then after some ms send packet B, do uWebsocket guarantees that we receive packet A then packet B? (not B then A)

**Comment by ghost** [comment]
Messages are ordered for send and for publish but not between send and publish (publishing is deferred to end of event loop iteration)
